src/traversal/traverser_core.rs

```diff
@@ -212,35 +219,40 @@ where
 
     // provided
 
-    fn iter_ptr_with_owned_storage<'a, V>(
+    fn iter_ptr_with_owned_storage<'t, V>(
         node_ptr: NodePtr<V>,
     ) -> impl Iterator<Item = <O::Enumeration as Enumeration>::Item<NodePtr<V>>>
     where
-        V: TreeVariant + 'a,
+        V: TreeVariant + 't,
+        <Self as TraverserCore<O>>::Storage<V>: 't,
     {
         Self::iter_ptr_with_storage(node_ptr, Self::Storage::default())
     }
 
-    fn iter_with_owned_storage<'a, V, M, P>(
-        node: &'a impl NodeRef<'a, V, M, P>,
-    ) -> impl Iterator<Item = OverItem<'a, V, O, M, P>>
+    fn iter_with_owned_storage<'t, 'a, V, M, P>(
+        node: &impl NodeRef<'a, V, M, P>,
+    ) -> impl Iterator<Item = OverItem<'a, V, O, M, P>> + 't
     where
         V: TreeVariant + 'a,
         M: MemoryPolicy,
         P: PinnedStorage,
+        Self::Storage<V>: 't,
+        'a: 't,
     {
         Self::iter_with_storage(node, Self::Storage::default())
     }
 
-    fn iter_mut_with_owned_storage<'a, V, M, P, MO>(
-        node_mut: &'a mut NodeMut<'a, V, M, P, MO>,
-    ) -> impl Iterator<Item = OverItemMut<'a, V, O, M, P>>
+    fn iter_mut_with_owned_storage<'t, 'a, V, M, P, MO>(
+        node_mut: &mut NodeMut<'a, V, M, P, MO>,
```

Comment:
```suggestion
        node: &mut NodeMut<'a, V, M, P, MO>,
```

The other functions practically all use `node`, so might be worth to change it here too to be consistent.